### ISSUE TITLE
[Bug] Findex panics when `ApplicationCommand::Command` variants `RString` is empty

### ISSUE THREAD

**Issue by Longestboi** [mdgaziur/findex#46]
## Description
When passing an empty `Rstring` to `ApplicationCommand::Command` in the return `RVec<FResult>` of a plugins `handle_query` function, Findex panics at [`crates/findex/src/gui/result_list_row.rs:136:42`](https://github.com/mdgaziur/findex/blob/8803441564ade854864ac16fc5938a7f03255dcf/crates/findex/src/gui/result_list_row.rs#L136-L136).

## Do you already have Findex Installed? (If yes, how did you install?)
I have installed Findex with the automated method detailed in [INSTALLATION.md](../blob/8803441564ade854864ac16fc5938a7f03255dcf/INSTALLATION.md).
Also, to test and attempt to fix the issue, I cloned the repo and ran Findex in release mode.

## To Reproduce
Return this from a plugins `handle_query` function:
```rust
RVec::from(vec![FResult {
    cmd: ApplicationCommand::Command(RString::new()),
    name: "No Command",
    desc: RNone,
    score: size::MAX,
    icon: RString::new()
}])
```

## Expected behavior
For the program to execute no command 

## System information
 - Distro: Ubuntu 22.04.2
 - Findex Version: development

## Output of `rustc --version --verbose` (if you compiled findex on your machine):
```
rustc 1.73.0 (cc66ad468 2023-10-03)
binary: rustc
commit-hash: cc66ad468955717ab92600c770da8c1601a4ff33
commit-date: 2023-10-03
host: x86_64-unknown-linux-gnu
release: 1.73.0
LLVM version: 17.0.2
```

## Additional context
To fix this issue, I added the below code above [result_list_row.rs:132](https://github.com/mdgaziur/findex/blob/8803441564ade854864ac16fc5938a7f03255dcf/crates/findex/src/gui/result_list_row.rs#L131-L131):
```Rust
if cmd.is_empty() {
    return;
}
```

I'm not sure if returning when the `cmd` `RString` is empty would be the most optimal solution. Adding a `None` variant to the `ApplicationCommand` enum or wrapping `cmd` in an `ROption` could be better options. Both would break plugins, either requiring recompilation or rewriting and recompilation, though.

**Comment by mdgaziur** [comment]
Hi, thanks for reporting this issue! I think, it'd be better to add a None variant to ApplicationCommand. That'll require minimal amount of changes in the codebase.

**Comment by mdgaziur** [comment]
I think this fix will open the door to making plugins like calculator, currency converter etc without Findex crashing every time the user presses enter.

**Comment by mdgaziur** [comment]
Should be fixed now in 46e977f
